fix: #4503

danking opened the pull request:
Fixes #4503 

Pco mistakenly used the number of values rather than the number of _valid_ values.

danking committed:
test: add a should_panic test (for now) for #4503

Signed-off-by: Daniel King <dan@spiraldb.com>

a10y commented on encodings/pco/src/test.rs:
```diff
@@ -42,6 +43,18 @@ fn test_compress_decompress() {
 
+#[test]
+#[should_panic]
+fn test_compress_decompress_big() {
+    let array = PrimitiveArray::from_option_iter([None, Some(1)]);
+    let compressed = PcoArray::from_primitive(&array, 3, 0).unwrap();
+    assert_eq!(compressed.scalar_at(0), Scalar::null_typed::<i32>());
+    assert_eq!(compressed.scalar_at(1), Scalar::from(Some(1)));
+    let decompressed = compressed.decompress();
+    assert_eq!(decompressed.scalar_at(0), Scalar::null_typed::<i32>());
+    assert_eq!(decompressed.scalar_at(1), Scalar::from(Some(1)));
```
Which one should panic? there are 4 different asserts here

danking replied:
It's actually the decompression that panics. I'll add a message check

danking replied:
Actually I found the fix.

danking committed:
specify the expected panic

Signed-off-by: Daniel King <dan@spiraldb.com>

danking committed:
fix: #4503

Fixes #4503.

Pco mistakenly used the number of values rather than the number of _valid_ values.

Signed-off-by: Daniel King <dan@spiraldb.com>
